lib/Server.js

```diff
@@ -2110,6 +2110,31 @@ class Server {
         middleware: this.setHeaders.bind(this),
       });
     }
+    {
+      /**
+       *
+       * @param {Request} req
+       * @param {Response} res
+       * @param {NextFunction} next
+       * @returns {void}
+       *
+       */
+      const optionsRequestResponseMiddleware = (req, res, next) => {
+        if (req.method === "OPTIONS") {
+          res.statusCode = 204;
+          res.setHeader("Content-Length", "0");
+          res.end();
+          return;
+        }
+        next();
+      };
+
+      middlewares.push({
+        name: "options-request-response",
+        path: "*",
+        middleware: optionsRequestResponseMiddleware,
+      });
```

Comment:
It can be problem, because we should not send response for any request...

Reply:
In my personal opinion. I thought it would be ok if we responded to OPTIONS requests immediately in the dev server...

Reply:
Do you want it for static assets or webpack assets?

Reply:
Both

Reply:
okay, implementation it for webpack assets is not a problem, with express static it can be problem due https://github.com/expressjs/serve-static/blob/master/index.js#L73... anyway I understand you problem, current solution is not good, so let's move it under an option, I think it will be better

Reply:
I forgot to mention that webpack-dev-server@4.6.0 process OPTIONS request correctly. During the time i didn't reply your comment(i'm sorry for that), I found the difference between 4.6.0 and latest, the change of serveMagicHtml(the last middleware/route) cause the different result, the commit hash is c13aa56. In the 4.6.0, it is implemented by using app.get, but in the latest version, it is implemented by using app.use. 
if we use app.use to implement it, this code will not be executed,
https://github.com/expressjs/express/blob/master/lib/router/index.js#L248
because `route` is undefined
https://github.com/expressjs/express/blob/master/lib/router/index.js#L232

Finally, back to express static, we can avoid this problem by moving the options-request-response middleware to the bottom of the middleware list. Anyway i agree with you that under an option is a better solution, but i think it's not necessary (just my stupid opinion).

Reply:
Yes, we change logic in magicHtml, because we want to remove it in the next major release, that is why we improve middlewares logic and now you can inject own middleware, I think your solution is good too, ideally - yes, we need handle it only for existing files, but it can be problem with express static, so currently I think we can move it under the option and improve it in future

Reply:
Uh,"under the options",is it mean that just move the code to the bottom of the options? I always thought the solution need with a devServer config, please forgive my poor english.

Reply:
Understood

Reply:
Finished